A few words about BC breaks (#179)
mautic/mautic-community-handbook: pages/15.contributing/02.developer/01.Code/02.Pull requests/docs.md

- Add unit tests to prove that the fixing of the bug or that the new feature actually works - see below.

### Backward compatibility  breaks

Try hard to not break backward compatibility (BC) - if you must do so, try to provide a compatibility layer to support the old way. PRs that break backward compatibility have less chance of acceptance, as they have to wait for release in a major release.

#### What is BC break?

Any change that may break a plugin eather using or extending a class. As Mautic has the plugin ecosystem so we must be careful even for a code we cannot see.

Examples:
- Removing or renaming a public or protected method in a non-final class. Create a new method instead and mark the old one deprecated.
- Changing the signature of a private or public method in a non-final class. Meaning adding, removing method parameters or adding or changing param or return types. Create a new method instead and mark the old one deprecated.

> - Changing behavior of a method so it does something else than before.

Reviewer: ```suggestion
- Changing behavior of a method so it does something differently.
```

- Adding a new method to an existing interface. Create a new interface instead.

> - Every time you are changing a TWIG template then think about the themes that are overwriting this template. For example changing the template name will cause issues.

Reviewer: ```suggestion
- Every time you change a TWIG template then think about the Themes that are overwriting this template. For example, changing the template name will cause issues.
```

#### What is not considered a BC break?

> On the other hand, there are some changes you can do and are not considered a BC break:

Reviewer: ```suggestion
On the other hand, there are some changes you can do that are not considered a BC break:
```

> - Changing constructor of a PHP service. Services are auto-wired so there is no harm in changing the dependencies.

Reviewer: ```suggestion
- Changing constructor of a PHP service. Services are autowired so there is no harm in changing the dependencies.
```

I _think_ it's only one word?
